src/main/java/org/engine/rest/UsersController.java

```diff
@@ -59,50 +65,210 @@
      * @return
      */
     @PostMapping("authorize_request")
-    public ResponseEntity<?> authorizeRequest(@RequestBody AuthenticationDTO resetDTO) {
-
+    public ResponseEntity<?> authorizeRequest(@Valid @RequestBody AuthenticationDTO resetDTO) {
+        // TODO - for remove - Spring Security should handle this request
         return null;
     }
 
+    // Reset password page
+
+    // Step 1 - from login page user enters e-mail to send new password
+
     /**
      * Endpoint used to send e-mail reset password request.
-     * @param resetDTO
+     * @param resetUserDTO
      * @return
      */
     @PostMapping("reset_request")
-    public ResponseEntity<?> resetRequest(@RequestBody AuthenticationDTO resetDTO) {
+    public ResponseEntity<?> resetRequest(@Valid @RequestBody ResetUserDTO resetUserDTO) {
 
-        return null;
+        return userService.findByLogin(resetUserDTO.getName()).map(user -> {
+
+            if (!user.getEmail().equals(resetUserDTO.getEmail())) {
+                return new ResponseEntity<>("NAME_AND_EMAIL_MISMATCH", HttpStatus.BAD_REQUEST);
```

Comment:
"NAME_AND_EMAIL_MISMATCH" constant to be centralised somewhere for reusability, maintainability reasons